script.js

```diff
@@ -0,0 +1,93 @@
+let defaultResult = 0;
+
+let currentResult = defaultResult;
+
+let logEntries = [];
+
+function getUserInput() {
+  return parseInt(userInput.value);
+}
+
+// the description & result output of calculation
+
+function createOutput(numberBeforeCalc, operator, calcNumber) {
+  const calcDescription = `${numberBeforeCalc} ${operator} ${calcNumber}`;
+  outputResult(currentResult, calcDescription);
+}
+
+// object inside array
+
+function writeToLog(
+  operationIdentifier,
+  prevResult,
+  numberEnteredByUser,
+  finalResult
+) {
+  const logEntry = {
+    operation: operationIdentifier,
+    PrevResult: prevResult,
+    number: numberEnteredByUser,
+    output: finalResult,
+  };
+
+  logEntries.push(logEntry);
+  console.log(logEntries);
+}
+
+// using if statement instead of seperate functions
+
+function calculationResult(calculationType) {
+  if (
+    (calculationType !== "addition" &&
+      calculationType !== "subtraction" &&
+      calculationType !== "multiplication" &&
+      calculationType !== "division") ||
+    getUserInput() === 0
+  ) {
+    return;
+  }
+
+  const initialResult = currentResult;
+  let mathOperator;
+
+  if (calculationType === "addition") {
+    currentResult = currentResult + getUserInput();
+    mathOperator = "+";
+  } else if (calculationType === "subtraction") {
+    currentResult = currentResult - getUserInput();
+    mathOperator = "-";
+  } else if (calculationType === "multiplication") {
+    currentResult = currentResult * getUserInput();
+    mathOperator = "*";
+  } else if (calculationType === "division") {
+    currentResult = currentResult / getUserInput();
+    mathOperator = "/";
+  }
+
+  createOutput(initialResult, mathOperator, getUserInput());
```

Comment:
instead of calling `getUserInput` everytime, save this in a `const` and use that const whenever you required an input from the user. @Rehan-Khan-Engineer

Reply:
Done @Abbas4tech